src/cpu/loongarch_utils.cpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * MegPeaK is Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * Copyright (c) 2021-2023 Megvii Inc. All rights reserved.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.
+ */
+
+#if MEGPEAK_LOONGARCH
+#include "src/cpu/loongarch_utils.h"
+
+using namespace megpeak;
+namespace {
+
+#define LA_HWCAP_LSX    (1 << 4)
+#define LA_HWCAP_LASX   (1 << 5)
+
+int feature_detect_lasx() {
+    int flag  = (int)getauxval(AT_HWCAP);
+
+    if (flag & LA_HWCAP_LASX)
+        return true;
+    else
+        return false;
+}
+
+int feature_detect_lsx() {
+    int flag  = (int)getauxval(AT_HWCAP);
+
+    if (flag & LA_HWCAP_LSX)
+        return true;
+    else
+        return false;
+}
+
+bool is_lasx_supported = feature_detect_lasx();
```

Comment:
it is better to put this global variable into the `is_supported` function as static data member

Reply:
Thank you for your advice, I will make the necessary modifications.